pkg/service/repair/helper_integration_test.go

```diff
@@ -0,0 +1,467 @@
+	ranges     []scyllaclient.TokenRange
+
+	// optional
+	SmallTableOptimization bool
+	RangesParallelism      int
+}
+
+func (r repairSchedReq) fullTable() string {
+	return r.keyspace + "." + r.table
+}
+
+type repairStatusReq struct {
+	host string
+	id   string
+}
+
+type repairSchedResp struct {
+	repairSchedReq
+	id string
+}
+type repairStatusResp struct {
+	repairStatusReq
+	status repairStatus
+}
+
+type repairStatus int
+
+const (
+	repairStatusDone repairStatus = iota
+	repairStatusFailed
+	repairStatusRunning
+)
+
+const (
+	oldRepairSchedPathPrefix  = "/storage_service/repair_async"
+	oldRepairStatusPathPrefix = "/storage_service/repair_status"
+	killRepairPathPrefix      = "/storage_service/force_terminate_repair"
+)
+
+func isOldRepairSchedReq(req *http.Request) bool {
+	return strings.HasPrefix(req.URL.Path, oldRepairSchedPathPrefix) && req.Method == http.MethodPost
+}
+
+func isRepairSchedReq(req *http.Request) bool {
+	return isOldRepairSchedReq(req)
+}
+
+func isOldRepairStatusReq(req *http.Request) bool {
+	return strings.HasPrefix(req.URL.Path, oldRepairStatusPathPrefix) && req.Method == http.MethodGet
+}
+
+func isRepairStatusReq(req *http.Request) bool {
+	return isOldRepairStatusReq(req)
+}
+
+func isKillRepairReq(req *http.Request) bool {
+	return strings.HasPrefix(req.URL.Path, killRepairPathPrefix) && req.Method == http.MethodPost
+}
+
+func newRepairSchedReq(t *testing.T, req *http.Request) (repairSchedReq, bool) {
+	if isOldRepairSchedReq(req) {
+		return newOldRepairSchedReq(t, req), true
+	}
+	return repairSchedReq{}, false
+}
+
+func newOldRepairSchedReq(t *testing.T, req *http.Request) repairSchedReq {
+	if !isOldRepairSchedReq(req) {
+		t.Error("Not old repair sched req")
+		return repairSchedReq{}
+	}
+
+	sched := repairSchedReq{
+		host:                   req.Host,
+		keyspace:               strings.TrimPrefix(req.URL.Path, oldRepairSchedPathPrefix+"/"),
+		table:                  req.URL.Query().Get("columnFamilies"),
+		replicaSet:             strings.Split(req.URL.Query().Get("hosts"), ","),
+		ranges:                 parseRanges(t, req.URL.Query().Get("ranges")),
+		SmallTableOptimization: req.URL.Query().Get("small_table_optimization") == "true",
+	}
+	if rawRangesParallelism := req.URL.Query().Get("ranges_parallelism"); rawRangesParallelism != "" {
+		rangesParallelism, err := strconv.Atoi(rawRangesParallelism)
+		if err != nil {
+			t.Error(err)
+			return repairSchedReq{}
+		}
+		sched.RangesParallelism = rangesParallelism
+	}
+	if sched.keyspace == "" || sched.table == "" || len(sched.replicaSet) == 0 {
+		t.Error("Not fully initialized old repair sched req")
+		return repairSchedReq{}
+	}
+
+	return sched
+}
+
+func newRepairStatusReq(t *testing.T, req *http.Request) (repairStatusReq, bool) {
+	if isOldRepairStatusReq(req) {
+		return newOldRepairStatusReq(t, req), true
+	}
+	return repairStatusReq{}, false
+}
+
+func newOldRepairStatusReq(t *testing.T, req *http.Request) repairStatusReq {
```

Comment:
I already left the comment under newOldRepairSchedReq which is valid to this function as well.